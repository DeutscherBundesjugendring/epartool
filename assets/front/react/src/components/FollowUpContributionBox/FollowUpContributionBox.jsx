import React from 'react';
import ArrayButton from '../ArrowButton/ArrowButton';


const FollowUpContributionBox = props => (
  <div className="followup-timeline-box">
    <div className="well well-bordered">
      <p>
        {props.contribution}
      </p>
      <div className="text-accent">
        {props.votingResults}
      </div>
    </div>
    <ArrayButton type="followup-timeline-count" label={props.childCount.toString()} />
  </div>
);

FollowUpContributionBox.propTypes = {
  contribution: React.PropTypes.string.isRequired,
  votingResults: React.PropTypes.string.isRequired,
  childCount: React.PropTypes.number.isRequired,
};

export default FollowUpContributionBox;
