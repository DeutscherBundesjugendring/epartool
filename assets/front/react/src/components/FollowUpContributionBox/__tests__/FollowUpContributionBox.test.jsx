import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import FollowUpContributionBox from '../FollowUpContributionBox';


describe('rendering', () => {
  it('renders correctly', () => {
    const tree = shallow(
      <FollowUpContributionBox contribution="Text" votingResults="Text" childCount={1} />
    );

    expect(shallowToJson(tree)).toMatchSnapshot();
  });;
});
