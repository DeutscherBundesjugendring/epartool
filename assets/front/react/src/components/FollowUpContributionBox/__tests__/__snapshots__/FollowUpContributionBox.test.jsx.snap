// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly 1`] = `
<div
  className="followup-timeline-box"
>
  <div
    className="well well-bordered"
  >
    <p>
      Text
    </p>
    <div
      className="text-accent"
    >
      Text
    </div>
  </div>
  <ArrowButton
    label="1"
    type="followup-timeline-count"
  />
</div>
`;
